Framework/Algorithms/src/RunCombinationHelpers/SampleLogsBehaviour.cpp

```diff
@@ -342,7 +342,7 @@ std::vector<double> SampleLogsBehaviour::createTolerancesVector(
 std::shared_ptr<Property> SampleLogsBehaviour::addPropertyForTimeSeries(
     const std::string &item, const double value, MatrixWorkspace &ws) {
   std::shared_ptr<Property> returnProp;
-  const std::string unit = ws.getLog(item)->units();
+  const std::string &unit = ws.getLog(item)->units();
```

Comment:
This change breaks the unit tests. The log unit is copied the `unit` as a backup here and restored at the end of this function. So the reference doesn't work here. Perhaps the variable could be renamed to `originalUnit` or similar to signify that the copy is really needed?

Reply:
That's quite obvious from the failing builds. I think a name change can prevent to think a reference could be possible. 👍